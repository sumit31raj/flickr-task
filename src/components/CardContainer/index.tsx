import React, { useEffect } from 'react';
import { useGetSearchImagesHook } from '../../services/flickr';

interface CardContainerProps {
  searchText: string;
}

const CardContainer = ({ searchText } : CardContainerProps) => {
  const { loading, data, setText: setSearchText, nextPage } = useGetSearchImagesHook();

  useEffect(() => {
    setSearchText(searchText);
  }, [searchText, setSearchText]);

  console.log('Loading:  ', loading);
  // console.log('Data:  ', data);

  return (
    <div className="CardContainer">
      Card Container!
      <button onClick={() => nextPage()}>Next Page!</button>
    </div>
  );
}

export default CardContainer;
