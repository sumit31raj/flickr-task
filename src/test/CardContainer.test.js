import { shallow } from 'enzyme';
import CardContainer from '../components/CardContainer/index';

describe('FeedContainer Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CardContainer searchText="" />);
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <InfiniteScrollComponent />', () => {
    expect(wrapper.find('InfiniteScroll')).toHaveLength(1);
  });
});
