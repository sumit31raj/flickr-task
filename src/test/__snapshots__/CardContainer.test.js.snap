// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedContainer Component renders 1`] = `ShallowWrapper {}`;
